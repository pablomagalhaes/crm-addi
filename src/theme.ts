import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  color: {
    white: '#FFF',
    black: '#424242',

    background: '#f2f2f2',
    border: 'rgba(0,0,0,0.1)',

    primary: '#6272a4',
    secondary: '#44475a',

    success: '#2AD178',
    warning: '#FABD59',
    danger: '#FF5B68',
  },

  font: {
    family: {
      primary: "'Roboto', sans-serif",
    },

    weight: {
      regular: 400,
      semiBold: 600,
    },

    color: {
      primary: '#424242',
      secondary: '#969696',
    },
  },

  borderRadius: {
    default: '3px',
  },

  breakpoint: {
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1300,
  },
};

export default theme;
