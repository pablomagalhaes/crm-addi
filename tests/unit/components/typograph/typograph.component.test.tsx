import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Title, Paragraph } from '@/components/typograph/typograph.component';
import theme from '@/theme';

describe('<Title />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Title theme={theme}>Some Title</Title>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the primary color as default', () => {
    const wrapper = shallow(<Title theme={theme} />);
    expect(wrapper).toHaveStyleRule('color', '#424242');
  });

  it('should have the primary color when is set', () => {
    const wrapper = shallow(<Title theme={theme} color="primary" />);
    expect(wrapper).toHaveStyleRule('color', '#424242');
  });

  it('should have the secondary color when is set', () => {
    const wrapper = shallow(<Title theme={theme} color="secondary" />);
    expect(wrapper).toHaveStyleRule('color', '#969696');
  });
});

describe('<Paragraph />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Paragraph theme={theme}>Some Text</Paragraph>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the primary color when is set', () => {
    const wrapper = shallow(<Paragraph theme={theme} color="primary" />);
    expect(wrapper).toHaveStyleRule('color', '#424242');
  });

  it('should have the secondary when is set', () => {
    const wrapper = shallow(<Paragraph theme={theme} color="secondary" />);
    expect(wrapper).toHaveStyleRule('color', '#969696');
  });
});
