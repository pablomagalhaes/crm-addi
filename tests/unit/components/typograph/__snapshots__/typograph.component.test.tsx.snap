// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should match snapchat 1`] = `
.c0 {
  font-family: primary:'Open Sans',sans-serif;
  color: #424242;
  font-size: 14px;
  line-height: 19px;
}

<p
  className="c0"
>
  Some Text
</p>
`;

exports[`<Title /> should match snapchat 1`] = `
.c0 {
  font-family: primary:'Open Sans',sans-serif;
  color: #424242;
  font-size: 32px;
  line-height: 42px;
}

<h2
  className="c0"
>
  Some Title
</h2>
`;
